packages/frontend/core/src/layouts/workspace-layout.tsx

```diff
@@ -145,6 +146,27 @@ export const WorkspaceLayoutInner = ({
   const { openPage } = useNavigateHelper();
   const pageHelper = usePageHelper(currentWorkspace.blockSuiteWorkspace);
   const t = useAFFiNEI18N();
+  const currentSyncEngineStatus = useCurrentSyncEngineStatus();
+
+  const haveUnsavedChanges = currentSyncEngineStatus?.local?.pendingPushUpdates;
```

Comment:
Based on `useCurrentSyncEngineStatus`, it seems the status is not a reactive value that will trigger React re-render on change.

Reply:
I change to useHaveUnsavedChanges, now only when haveUnsavedChanges changes between true or false will trigger rerender